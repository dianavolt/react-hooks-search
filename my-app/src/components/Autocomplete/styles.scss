.input-suggestions {
    padding: 10px;
    border: 1px solid grey;
    border-radius: 2px;
    margin-top: 2px;
    width: 100%;
    width:200px;
    right: 0;
    left: 0;
    box-sizing: border-box;
    background: #ffffff;
    text-align: left;
    z-index: 2;

    .search-link {
        display: block;
        margin-top: 10px;
        padding: 4px 0;
    }
}