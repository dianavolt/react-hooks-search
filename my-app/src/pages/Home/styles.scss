.home-page-container {
    // background: #f6f6f6;
    text-align: center;
    margin: 0 auto;
    padding-top: 200px;
    overflow: hidden;
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height:100%;
    max-width: 100%;
    .img {
        max-width: 200px;
        margin-bottom: 40px;
    }
}