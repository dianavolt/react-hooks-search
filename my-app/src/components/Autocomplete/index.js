import React from 'react'
import ReactAutocomplete from 'react-autocomplete'
import PropTypes from 'prop-types'

import Input from '../Input'

import './styles.scss'

const Autocomplete = ({ articles, searchVal, onSearchChange }) => (
    <ReactAutocomplete
        items={articles}
        renderInput={Input}
        inputProps={{placeholder: 'Enter search term...'}}
        getItemValue={item => item.label}
        renderMenu={(children, value, style) => {
            return (articles && articles.length > 0)?
                (<div style={{...style}} className="input-suggestions">
                    {children}
                    <a href={`/search?query=${value}`} className='search-link'>
                        See all results
                    </a>
                </div>
                ) : <></>
        }}
        renderItem={(item, highlighted) =>
            <div
                key={item.id}
                style={{ backgroundColor: highlighted ? '#eee' : 'transparent'}}
            >
                <a href={item.id} target="_blank">{item.label}</a>
            </div>
        }
        value={searchVal}
        onChange={onSearchChange}
    />
)

Autocomplete.propTypes = {
    articles: PropTypes.array,
    searchVal:PropTypes.string
}

export default Autocomplete
