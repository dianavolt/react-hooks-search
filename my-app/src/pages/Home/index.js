import React from 'react'
import Container from '../../components/Container'
import Autocomplete from '../../components/Autocomplete'

import './styles.scss'
import logo from '../../assets/Wikipedia-logo.png'

const Home = () => {
    return (
        <div className='home-page-container'>
            <img alt='logo' src={logo} />
            <Container>
                {({ articles, searchVal, onSearchChange }) => 
                <Autocomplete 
                    articles={articles}
                    searchVal={searchVal}
                    onSearchChange={onSearchChange}
                />
                }
            </Container>
        </div>
    )
}

export default Home
