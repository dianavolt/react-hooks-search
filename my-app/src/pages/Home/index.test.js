import React from 'react'
import { mount } from 'enzyme'

import Home from '.'
import Autocomplete from '../../components/Autocomplete'

const render = (props, children) => mount(<Home {...props} >{children}</Home>)

describe('Home component', () => {
    let test
    let props    
    let children = ({searchVal, onSearchChange, articles}) => <Autocomplete searchVal={searchVal}
                                                                            onSearchChange={onSearchChange}
                                                                            articles={articles}/>

    describe('without props', () => {
        beforeEach(() => {
            test = render(props, children)
        })

        it('should match snapshot', () => {
            expect(test).toMatchSnapshot()
        })
    })
})
