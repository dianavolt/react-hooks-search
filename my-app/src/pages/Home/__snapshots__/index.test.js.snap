// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component without props should match snapshot 1`] = `
<Home>
  <div
    className="home-page-container"
  >
    <img
      alt="logo"
      src="Wikipedia-logo.png"
    />
    <Container>
      <Autocomplete
        articles={Array []}
        onSearchChange={[Function]}
        searchVal=""
      >
        <Autocomplete
          autoHighlight={true}
          getItemValue={[Function]}
          inputProps={
            Object {
              "placeholder": "Enter search term...",
            }
          }
          isItemSelectable={[Function]}
          items={Array []}
          menuStyle={
            Object {
              "background": "rgba(255, 255, 255, 0.9)",
              "borderRadius": "3px",
              "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
              "fontSize": "90%",
              "maxHeight": "50%",
              "overflow": "auto",
              "padding": "2px 0",
              "position": "fixed",
            }
          }
          onChange={[Function]}
          onMenuVisibilityChange={[Function]}
          onSelect={[Function]}
          renderInput={[Function]}
          renderItem={[Function]}
          renderMenu={[Function]}
          selectOnBlur={false}
          value=""
          wrapperProps={Object {}}
          wrapperStyle={
            Object {
              "display": "inline-block",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              className="input-field"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter search term..."
              role="combobox"
              value=""
            />
          </div>
        </Autocomplete>
      </Autocomplete>
    </Container>
  </div>
</Home>
`;
